cdblib/cdblib.py

```diff
@@ -15,34 +15,79 @@
 
 from .djb_hash import djb_hash
 
-
+# Structs for 32-bit databases
 read_2_le4 = Struct('<LL').unpack
 read_2_le8 = Struct('<QQ').unpack
+
+# Structs for 64-bit databases
 write_2_le4 = Struct('<LL').pack
 write_2_le8 = Struct('<QQ').pack
 
+# Encoders for keys
+DEFAULT_ENCODERS = {six.text_type: lambda x: x.encode('utf-8')}
+for t in six.integer_types:
+    DEFAULT_ENCODERS[t] = lambda x: six.text_type(x).encode('utf-8')
+
+
+class _CDBBase(object):
+    def __init__(self, hashfn=djb_hash, strict=False, encoders=None):
+        self.hashfn = hashfn
+
+        if strict:
```

Comment:
We only check `strict`-ness at instantiation time, rather than on each insert or query.